### PR TITLE
[FLINK-9020][E2E Tests] Use separate modules per testcase

### PR DESCRIPTION
**Pull request by florianschmidt1994**
## Use separate maven modules per test case in end-to-end tests
This change sets up the end-to-end project dir in a way that each test that provides its own sources does so with a separate module.
The main motivation is that now each test case can easier specify the way that it is supposed to be built, which could often for example be based on the quickstart example. There then is no need anymore to put together custom packaging instructions for each addition that is added to the test cases.

## Brief change log
  - Change parent artifact from jar to pom packaging
  - Rename parent artifact from  `flink-end-to-end-tests_${scala.binary.version}` to `flink-end-to-end-tests` (needed for having a successful build)
  - Move existing ClassLoader PoC to its own module
  - Fix paths in test scripts

## Verifying this change
* I reran the tests by hand and they still behave the same

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by florianschmidt1994** [comment]
@kl0u Could you have a look at this?

File: flink-end-to-end-tests/parent-child-classloading-test/pom.xml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	<parent>
+		<artifactId>flink-end-to-end-tests</artifactId>
+		<groupId>org.apache.flink</groupId>
+		<version>1.6-SNAPSHOT</version>
+		<relativePath>..</relativePath>
+	</parent>
+
+	<modelVersion>4.0.0</modelVersion>
+
+	<artifactId>parent-child-classloading-test</artifactId>
```
Comment (zentol) [inline flink-end-to-end-tests/parent-child-classloading-test/pom.xml]: * add `flink` prefix
* add `_${scala.binary.version}` suffix
* add `<name>flink-parent-child-classloading-test</name>` (this is what allows us to omit the scala stuff when listing child modules)

**Comment by florianschmidt1994** [comment]
Thanks @zentol! I addressed your comments and will squash if approved

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**